### PR TITLE
feat: add fetching of all parcels

### LINKED ISSUE #9: Fetch the relevant parcel data from the API
We need to get parcel data from https://discordapp.com/channels/@me/1197621574199685251/1200181459059748884 so that we can propagate this data further to the view components and then post them to the end user.
This issue is specific to the fetching and parsing of the data.

Todo:
- [x] Read documentation regarding api fetching in nextjs
- [x] Determine relevant data to read
- [x] Fetch the data
- [x] Type check the data

### PR DESCRIPTION
**Pull request by EmberCraze**
We need to get all the parcels so we can display them all to the end user!

The type is according to the url resource at https://my.api.mockaroo.com/orders.json?key=e49e6840

### CODE REVIEW

**Comment by EmberCraze** [comment]
This is a general comment

File: src/utils/parcel-fetch.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export type Parcel= {
+  id: 1
```
Comment (r8now) [inline src/utils/parcel-fetch.ts]: id is hardcoded to the number 1, maybe better to set it to number.
id: number,

File: src/utils/parcel-fetch.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  location_coordinate_latitude: number
+  location_coordinate_longitude: number
+  location_status_ok: boolean
+  user_phone: string
+  user_name: string
+  notes?: string
+  last_updated: string
+}
+
+export async function getAllParcelData(): Promise<Array<Parcel>> {
```
Comment (r8now) [inline src/utils/parcel-fetch.ts:19]: Good to add a try catch block here.

export async function getAllParcelData(): Promise<Array<Parcel>> {
  try {
    const res = await fetch('https://my.api.mockaroo.com/orders.json?key=e49e6840');
    if (!res.ok) {
      throw new Error('Failed to fetch parcels');
    }
    return res.json();
  } catch (error) {
    console.error("Error fetching parcel data:", error);
    return [];
  }
}

File: src/utils/parcel-fetch.ts (hunk 1 above)
Comment (EmberCraze) [reply src/utils/parcel-fetch.ts]: Nice catch missed that. Will fix

File: src/utils/parcel-fetch.ts (hunk 2 above)
Comment (EmberCraze) [reply src/utils/parcel-fetch.ts:19]: Good feedback, will fix this

Comment (EmberCraze) [reply src/utils/parcel-fetch.ts:19]: Fixed

File: src/utils/parcel-fetch.ts (hunk 1 above)
Comment (EmberCraze) [reply src/utils/parcel-fetch.ts]: Fixed

File: src/utils/parcel-fetch.ts (hunk 2 above)
Comment (r8now) [reply src/utils/parcel-fetch.ts:19]: niceee

**Review by r8now — APPROVED** [review]
looks good

**Review by r8now — APPROVED** [review]
Good

**Review by r8now — COMMENTED** [review]
good

**Pull request merged**